### DISCUSSION TITLE
Best way to set all episode before a given one as watched except those already watched by the user ?

### DISCUSSION THREAD

**Discussion by jonathanantoine** [trakt/trakt-api#282] [Q&A] [answered]
Hello,

What is the best way to set all episode before a given one as watched except those already watched by the user ? 

Currently I fetch all the already watched once and filter them out by hand but I would prefer a server side method as sometime the api data is not up to date with the website.

Thanks !

**Comment by rectifyer** [comment, marked as answer]
We don't have a way to do this server side yet, but that might be a good update. I'll look into this and see if we can support it without a performance hit. The issue with watched history is basically any additional checks make the performance that much worse. The larger issue is a sync method can accept multiple shows, so if you sent 100 shows and each one needed to "mark up to episode X" that is ton of extra processing. I'm thinking this will be more of a Trakt VPI v3 feature.
